File: demo/src/main/java/ma/toubkalit/suiviprojet/services/FactureService.java
package ma.toubkalit.suiviprojet.services;

import ma.toubkalit.suiviprojet.dto.facture.FactureRequestDto;
import ma.toubkalit.suiviprojet.dto.facture.FactureResponseDto;

import java.util.List;

public interface FactureService {

    FactureResponseDto create(Long phaseId, FactureRequestDto requestDto);

    List<FactureResponseDto> getAll();

    FactureResponseDto getById(Long id);

    FactureResponseDto update(Long id, FactureRequestDto requestDto);

    void delete(Long id);
}

File: demo/frontend/src/services/modules/factureService.js
import api from '../api';

export const factureService = {
  // GET /api/factures
  getAll: async () => {
    console.log(`[factureService] GET /factures`);
    const response = await api.get(`/factures`);
    return response.data;
  },
  
  // GET /api/factures/{id}
  getById: async (id) => {
    console.log(`[factureService] GET /factures/${id}`);
    const response = await api.get(`/factures/${id}`);
    return response.data;
  },
  
  // POST /api/phases/{phaseId}/facture
  create: async (phaseId, data) => {
    console.log(`[factureService] POST /phases/${phaseId}/facture`, data);
    const response = await api.post(`/phases/${phaseId}/facture`, data);
    return response.data;
  },
  
  // PUT /api/factures/{id}
  update: async (id, data) => {
    console.log(`[factureService] PUT /factures/${id}`, data);
    const response = await api.put(`/factures/${id}`, data);
    return response.data;
  }
};
